lib/sendgrid_actionmailer.rb

```diff
@@ -119,6 +108,12 @@ def get_disposition(message)
       content_disp.disposition_type
     end
 
+    def add_api_key(sendgrid_mail, mail)
+      if mail['delivery-method-options'] && mail['delivery-method-options'].value.include?('api_key')
```

Comment:
Why not set it to be `settings.fetch(:api_key)` first, before the iff statement, then the `key = api_key || ..` line above could be removed.

Reply:
```
def add_api_key(sendgrid_mail, mail)
  self.api_key = settings.fetch(:api_key)
  if mail['delivery-method-options'] && mail['delivery-method-options'].value.include?('api_key')
    self.api_key = JSON.parse(mail['delivery-method-options'].value.gsub('=>', ':'))['api_key']
  end
end
```

Reply:
Also, should probably raise an error here at the end if `self.api_key` is undefined?

Reply:
Well, my thinking is we need to maintain two sets of keys, correct?  A global api key and a request api_key?  If a request api_key is provided, I don't think we want the global to be updated. The next request without a request api_key key should revert to the global stored in settings.

Reply:
Actually, now that I look this over again, this wouldn't work as desired.  The request api_key would not be temporary and would persist in client.

Reply:
@tyrauber If you set it to be `nil` first, in `add_api_key`, this will work just fine. And `settings.fetch(:api_key)` will return `nil` if `api_key` isn't found.

Reply:
The function above should cover it, but if you want to do `self.api_key = settings.fetch(:api_key, nil)`, to be more explicit about the intent?